urlshort/handler.go

```diff
@@ -33,5 +42,24 @@ func MapHandler(pathsToUrls map[string]string, fallback http.Handler) http.Handl
 // a mapping of paths to urls.
 func YAMLHandler(yml []byte, fallback http.Handler) (http.HandlerFunc, error) {
 	// TODO: Implement this...
-	return nil, nil
+	parsedYaml, err := parseYAML(yml)
+	if err != nil {
+		return nil, err
+	}
+	pathMap := buildMap(parsedYaml)
+	return MapHandler(pathMap, fallback), nil
+}
+
+func parseYAML(yml []byte) (out []map[string]string, err error) {
+	err = yaml.Unmarshal(yml, &out)
+	return out, err
+}
+
+func buildMap(parsedYaml []map[string]string) map[string]string {
```

Comment:
is this going to only accept YAML? The bonus objective is to also be able to use JSON.